auth/src/main/java/com/firebase/ui/auth/ui/email/CheckEmailFragment.java

```diff
@@ -100,9 +101,11 @@ public View onCreateView(LayoutInflater inflater,
         View v = inflater.inflate(R.layout.check_email_layout, container, false);
 
         // Email field and validator
+        mEmailLayout = (TextInputLayout) v.findViewById(R.id.email_layout);
         mEmailEditText = (EditText) v.findViewById(R.id.email);
-        mEmailFieldValidator = new EmailFieldValidator(
-                (TextInputLayout) v.findViewById(R.id.email_layout));
+        mEmailFieldValidator = new EmailFieldValidator(mEmailLayout);
+        mEmailLayout.setOnClickListener(this);
```

Comment:
Because the error text does not clear once the EditText is clicked. I felt that if the user clicks this field, the error message should reset. Thus I implemented the click listener and set the error to null.